### PR TITLE
Asynchronicity improvements

### PR DESCRIPTION
**Pull request by MathieuDutSik**

### CODE REVIEW

File: linera-chain/src/chain.rs [hunk 1]
```diff
@@ -556,6 +556,7 @@ where
     where
         F: Fn(E) -> Message,
     {
+        use futures::stream::{self, StreamExt, TryStreamExt};
```
Comment (afck) [inline linera-chain/src/chain.rs:559]: Why not include it globally, with the other imports?

File: linera-chain/src/chain.rs [hunk 2]
```diff
@@ -641,22 +648,37 @@ where
...
+                Ok::<Option<(Target, BlockHeight)>, ChainError>(result)
+            })
+            .buffer_unordered(C::MAX_CONNECTIONS);
+        let infos = stream.try_collect::<Vec<_>>().await?;
+        let mut targets = Vec::new();
+        let mut heights = Vec::new();
+        for (target, height) in infos.into_iter().flatten() {
+            targets.push(target);
+            heights.push(height);
+        }
```
Comment (afck) [inline linera-chain/src/chain.rs:677]: I think this is `Iterator::unzip`:

```suggestion
        let (targets, heights): (Vec<_>, Vec<_>) = infos.into_iter().flatten().unzip();
```

**Review by afck — APPROVED** [review]

File: linera-chain/src/chain.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-chain/src/chain.rs:559]: Indeed.

**Review by afck — APPROVED** [review]

**Review by ma2bd — COMMENTED** [review]
Thanks for the improvements!

**Pull request merged**